operator/internal/controller/redpanda/nodepool_controller.go

```diff
@@ -92,5 +111,58 @@ func (r *NodePoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (r
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	var status statuses.NodePoolStatus
+	sts, err := statefulSetForNodePool(ctx, r.Client, pool)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if sts == nil {
+		status.SetDeployed(statuses.NodePoolDeployedReasonNotDeployed)
+	}
+
+	originalPoolStatus := pool.Status.EmbeddedNodePoolStatus
+	pool.Status.EmbeddedNodePoolStatus = redpandav1alpha2.EmbeddedNodePoolStatus{}
+
+	if sts != nil {
+		desiredReplicas := ptr.Deref(pool.Spec.Replicas, 3)
+		condemnedReplicas := sts.Status.Replicas - desiredReplicas
+		if condemnedReplicas < 0 {
+			condemnedReplicas = 0
+		}
+
+		if desiredReplicas == sts.Status.Replicas {
+			status.SetDeployed(statuses.NodePoolDeployedReasonDeployed)
+		} else {
+			status.SetDeployed(statuses.NodePoolDeployedReasonScaling)
+		}
+
+		pool.Status.EmbeddedNodePoolStatus = redpandav1alpha2.EmbeddedNodePoolStatus{
```

Comment:
There's no acknowledgment of the STS being up to date here. If the Redpanda control loop happens to get wedged (or if there's an unfortunate timing), this could produce unreliable or confusing results.

The best solution I can think of it passing the generation of the nodepool through to the STS (not the Pods) as a label and using that to fill in ObservedGeneration here.

I don't think we need to fix it unless we plan on waiting on the status of nodepools in cloud but it would be good to note (or tell me I'm wrong).

Reply:
That shouldn’t be too bad to plumb through, I’ll give it a shot

Reply:
done